packages/fuels-core/src/types/wrappers/transaction.rs

```diff
@@ -451,7 +480,9 @@ macro_rules! impl_tx_wrapper {
                         _ => {
                             // not a contract, it's safe to unwrap
                             let owner = extract_owner_or_recipient(input).unwrap();
-                            let asset_id = input.asset_id().unwrap().to_owned();
+                            // Can we just use `BASE_ASSET_ID` here ? cc @AhmedM
```

Comment:
From looking at the fuel-core impl of `asset_id` it seems that the arguments is used the return the asset id of message coins which according to protocol is always the baset asset. 
So I would say yes, but I wonder why it wasn't hardcoded on the core side. @xgreenx should know

Reply:
In the future, we plan not to support different base asset ids.